lambda/shared/src/main/scala/feral/lambda/events/SnsEvent.scala

```diff
@@ -0,0 +1,223 @@
+      "EventSubscriptionArn": "arn:aws:sns:TEST",
+      "EventSource": "aws:sns",
+      "Sns": {
+        "Signature": "TEST",
+        "MessageId": "9f0cfa95-e344-4f93-89ab-5e979503ed1f",
+        "Type": "Notification",
+        "TopicArn": "arn:aws:sns:TEST",
+        "MessageAttributes": {
+          "string": {
+            "Type": "String",
+            "Value": "Testing"
+          },
+          "binary": {
+            "Type": "Binary",
+            "Value": "Testing"
+          },
+          "number": {
+            "Type": "Number",
+            "Value": 1e6
+          },
+          "numberString": {
+            "Type": "Number",
+            "Value": "5"
+          },
+          "array": {
+            "Type": "String.Array",
+            "Value": "[1, \"two\", true, false]"
+          },
+          "unsupported": {
+            "Type": "FancyNewType",
+            "Value": "SpecialValueHere"
+          }
+        },
+        "SignatureVersion": "1",
+        "Timestamp": "2022-04-06T01:02:03.456Z",
+        "SigningCertUrl": "TEST",
+        "Message": "Testing Message",
+        "UnsubscribeUrl": "TEST",
+        "Subject": "Test Message"
+      }
+    }
+  ]
+}
+
+==>
+
+SnsEvent(
+  List(
+    SnsRecord(
+      "1.0",
+      "arn:aws:sns:TEST",
+      "aws:sns",
+      SnsMessage(
+        "TEST",
+        9f0cfa95-e344-4f93-89ab-5e979503ed1f,
+        "Notification",
+        "arn:aws:sns:TEST",
+        Map(
+          "number" -> Number(1E+6),
+          "binary" -> Binary(ByteVector(5 bytes, 0x4deb2d8a78)),
+          "string" -> String("Testing"),
+          "numberString" -> Number(5),
+          "array" -> StringArray(List(Number(1), String("two"), Boolean(true), Boolean(false))),
+          "unsupported" -> Unknown("FancyNewType", Some("SpecialValueHere"))
+        ),
+        "1",
+        2022-04-06T01:02:03.456Z,
+        "TEST",
+        "Testing Message",
+        "TEST",
+        "Test Message"
+      )
+    )
+  )
+)
+ */
+
+final case class SnsEvent(
+    records: List[SnsEventRecord]
+)
+
+object SnsEvent {
+  implicit val decoder: Decoder[SnsEvent] =
+    Decoder.forProduct1("Records")(SnsEvent.apply)
+}
+
+final case class SnsEventRecord(
+    eventVersion: String,
+    eventSubscriptionArn: String,
+    eventSource: String,
+    sns: SnsMessage
+)
+
+object SnsEventRecord {
+  implicit val decoder: Decoder[SnsEventRecord] = Decoder.forProduct4(
+    "EventVersion",
+    "EventSubscriptionArn",
+    "EventSource",
+    "Sns"
+  )(SnsEventRecord.apply)
+}
+
+final case class SnsMessage(
+    signature: String,
+    messageId: UUID,
+    `type`: String,
+    topicArn: String,
+    messageAttributes: Map[String, SnsMessageAttribute],
+    signatureVersion: String,
+    timestamp: Instant,
+    signingCertUrl: String,
+    message: String,
+    unsubscribeUrl: String,
+    subject: String
+)
+
+object SnsMessage {
+  implicit val instantDecoder: Decoder[Instant] = Decoder.decodeString.emapTry { str =>
```

Comment:
Is this not in Circe? I think they have date/time instances.

Reply:
Quite correct. Fixed.